course5/homework05/vector2d.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+#include <iostream>
+#include <string>
+
+namespace vector
+{
+class Vector2D
+{
+public:
+    enum class System { CARTESIAN, POLAR };
+
+    Vector2D(double initX = 0.0, double initY = 0.0);
+    Vector2D(double arg1, double arg2, System system);
+
+    double GetX() const;
+    double GetY() const;
+
+    void SetX(double newX);
+    void SetY(double newY);
+
+    double GetR() const;
+    double GetTheta() const;
+
+    std::string AsPolar() const;
+
+    Vector2D operator+(const Vector2D& vec) const;
+    Vector2D operator-() const;
+    Vector2D operator-(const Vector2D& vec) const;
+    void operator+=(const Vector2D& vec);
+    void operator-=(const Vector2D& vec);
+    bool operator==(const Vector2D& vec) const;
+    bool operator!=(const Vector2D& vec) const;
+
+private:
+    double x;
+    double y;
+};
+
+}
+
+vector::Vector2D operator "" _x(long double initX);
+vector::Vector2D operator "" _y(long double initY);
+
+vector::Vector2D operator "" _r(long double initR);
+vector::Vector2D operator "" _theta(long double initTheta);
```

Comment:
Taking them out and adding the qualifiers was what fixed the rest of my linker errors. If I put them inside the namespace, I get
```
Undefined symbols for architecture x86_64:
  "vector::operator"" _r(long double)", referenced from:
      _main in main.o
  "vector::operator"" _x(long double)", referenced from:
      _main in main.o
  "vector::operator"" _y(long double)", referenced from:
      _main in main.o
  "vector::operator"" _theta(long double)", referenced from:
      _main in main.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [vector2d.out] Error 1
make[1]: *** [CMakeFiles/vector2d.out.dir/all] Error 2
make: *** [all] Error 2
```